be/tool/beck.c

```diff
@@ -1156,8 +1158,9 @@ static void emap_act(struct action *act, struct m0_be_tx *tx)
 	emap_val = emap_ac->emap_val.b_addr;
 	if (emap_val->er_value != AET_HOLE) {
 		emap_key = emap_ac->emap_key.b_addr;
-		ext.e_start = emap_val->er_start >> adom->sad_babshift;
-		ext.e_end =  emap_key->ek_offset >> adom->sad_babshift;
+		ext.e_start = emap_val->er_value >> adom->sad_babshift;
+		ext.e_end   = (emap_val->er_value + emap_key->ek_offset -
+			       emap_val->er_start) >> adom->sad_babshift;
```

Comment:
[* defect *] The logic has changed, and there is no mention of why it has changed anywhere. Please describe in the commit message.

The values passed to balloc are different now, and I don't see an explanation about why new version would work.

Reply:
Updated commit description

Reply:
Please describe how this change was tested. I'd like to see a test description for a test that was failing before, but which passes now.

Reply:
This change is tested using https://jts.seagate.com/browse/TEST-13362, and 
now we are not getting error message **"Failed to reserve extent"**